exp/views/responses.py

```diff
@@ -1109,6 +1300,49 @@ def get(self, request, *args, **kwargs):
         return response
 
 
+class StudyResponsesFrameDataPsychDS(ResponseDownloadMixin, generic.list.ListView):
+    """Hitting this URL downloads a ZIP file in Psych-DS formatting with frame data from one response per file in CSV format"""
+
+    def render_to_response(self, context, **response_kwargs):
+        paginator = context["paginator"]
+        study = self.study
+        study_short_uuid = study.uuid.hex[:4]
+
+        if study.study_type.is_external:
+            messages.error(
+                self.request, "Frame data is not available for External Studies."
+            )
+
+            return study_responses_all(study)
```

Comment:
Nice! At first I was confused about this because I tried testing it and saw that there's no "Psych-DS formatted frame data" option on the response downloads page. But then I realized that I could still go directly to the URL, so I tried that and got the banner message on the top of the downloads page 👍